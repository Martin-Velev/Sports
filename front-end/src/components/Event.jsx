import React from "react";
import { Link } from "react-router-dom";

const Event = props => {
  const { sportId, event } = props;
  if (!sportId || !event) {
    throw new Error("Incorrect props");
  }
  return (
    <Link to={`/sports/${sportId}/events/${event.id}`}>
      {event.description}
    </Link>
  );
};

export default Event;
