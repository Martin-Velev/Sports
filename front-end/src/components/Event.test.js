import React from "react";
import renderer from "react-test-renderer";
import Event from "../components/Event";
import { BrowserRouter as Router } from "react-router-dom";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <Event event={{ id: 1, description: "test Event" }} sportId={42} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
