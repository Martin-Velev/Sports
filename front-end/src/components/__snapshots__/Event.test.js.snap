// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="/sports/42/events/1"
  onClick={[Function]}
>
  test Event
</a>
`;
